backstory/src/test/java/com/google/sps/PerspectiveManagerTest.java

```diff
@@ -0,0 +1,187 @@
+    
+    Assert.assertEquals("foo", manager.getText());
+
+    Map<AttributeType, Float> output = manager.getAnalyses();
+    for(AttributeType type: output.keySet()) {
+      Assert.assertEquals(-1f, output.get(type), 0);
+    }
+
+    // -1 of toxicity is < .7 (the current rule) so should be considered appropriate
+    Assert.assertEquals(true, manager.getDecision());
+  }
+
+  @Test 
+  public void checkSizeOfMap() {
+    // Check that PerspectiveManager analyzes and returns all 15 of our desired attributes
+    Map<AttributeType, Float> input = new HashMap<AttributeType, Float>();
+    PerspectiveAPI api = setUpMockAPI(input);
+
+    PerspectiveManager manager = new PerspectiveManager(api, "foo");
+    Map<AttributeType, Float> output = manager.getAnalyses();
+
+    Assert.assertEquals(15, output.size());
+  }
+
+  @Test
+  public void checkDecisionBasedOnToxicity() {
+    // Check that PerspectiveManager makes the right decision when given 
+    // inputs with different levels of toxicity). getDecision() should return
+    // true if toxicity < .7 and false if toxicity >= .7
+    
+    Map<AttributeType, Float> input = new HashMap<AttributeType, Float>();
+    PerspectiveAPI api;
+
+    // check that if below .7, it returns true
+    final float LOW_TOXICITY = .69f;
+    input.put(AttributeType.TOXICITY, LOW_TOXICITY);
+    api = setUpMockAPI(input);
+
+    PerspectiveManager lowToxicity = new PerspectiveManager(api, "foo");
+    
+    Assert.assertEquals(LOW_TOXICITY, lowToxicity.getAnalyses().get(AttributeType.TOXICITY), 0);
+    Assert.assertEquals(true, lowToxicity.getDecision());
+
+    // check that if equal to .7, it returns false
+    final float EQUAL_TOXICITY = .7f;
+    input.put(AttributeType.TOXICITY, EQUAL_TOXICITY);
+    api = setUpMockAPI(input);
+
+    PerspectiveManager equalToxicity = new PerspectiveManager(api, "foo");
+    
+    Assert.assertEquals(EQUAL_TOXICITY, equalToxicity.getAnalyses().get(AttributeType.TOXICITY), 0);
+    Assert.assertEquals(false, equalToxicity.getDecision());
+
+    // check that if above .7, it returns false
+    final float HIGH_TOXICITY = .71f;
+    input.put(AttributeType.TOXICITY, HIGH_TOXICITY);
+    api = setUpMockAPI(input);
+
+    PerspectiveManager highToxicity = new PerspectiveManager(api, "foo");
+    
+    Assert.assertEquals(HIGH_TOXICITY, highToxicity.getAnalyses().get(AttributeType.TOXICITY), 0);
+    Assert.assertEquals(false, highToxicity.getDecision());
+  }
+
+  /** 
+   * Returns a mock version of the PerspectiveAPI returns the scores that you set in desiredScores
+   * or which returns -1 when AttributeType scores are set.
+   *
+   * @param desiredScores a map of the scores you want the mockAPI to return 
+   *    (if you don't set any, the mock version of this API will always return -1)
+   * @return the mocked version of the PerspectiveAPI
+   */
+  private static PerspectiveAPI setUpMockAPI(Map<AttributeType, Float> desiredScores) {
```

Comment:
There's a JUnit @Before tag used in the tdd section that can be to reduce potential redundancies and automatically handle some setup versus recalling this.

Reply:
Unfortunately, you have to set up a new mock API every time you want custom values, which could be different each test. That's why it's called every time. As such, I think that creating one mock API at the beginning wouldn't be that helpful. @UdayKalra Do you have a different suggestion as to how to reduce redundancy?

Reply:
You can establish api and input as class fields and reinstantiate each test versus redeclaring the typing every test.

Reply:
Also this " Map<AttributeType, Float> input = new HashMap<AttributeType, Float>();" can go in a @before

Reply:
Done

Reply:
Thank you. That's super helpful. I will fix these.

Reply:
Done